playbooks/openstack-osh_instance.yml

```diff
@@ -0,0 +1,89 @@
+    - name: Load openstack vars
+      include_vars: "{{ playbook_dir }}/../vars/deploy-on-openstack.yml"
+
+    - name: Handle creation
+      when:
+        - not ((osh_node_delete | default('false'))| bool)
+      block:
+        - name: Create volume
+          os_volume:
+            cloud: "{{ deploy_on_openstack_cloudname }}"
+            display_name: "{{ deploy_on_openstack_oshnode_volume_name }}"
+            size: 80
+            state: "present"
+        - name: Create server
+          os_server:
+            cloud: "{{ deploy_on_openstack_cloudname }}"
+            flavor: "{{ deploy_on_openstack_oshnode_flavor }}"
+            image: "{{ deploy_on_openstack_oshnode_image }}"
+            key_name: "{{ deploy_on_openstack_keypairname }}"
+            name: "{{ deploy_on_openstack_oshnode_server_name }}"
+            network: "{{ deploy_on_openstack_internal_network }}"
+            security_groups:
+              - "{{ deploy_on_openstack_oshnode_securitygroup }}"
+            auto_ip: True
+            state: present
+            timeout: 300
+            volumes:
+              - "{{ deploy_on_openstack_oshnode_volume_name }}"
+          register: _osh_server_create
+        - name: Show the server data
+          debug:
+            var: _osh_server_create
+        - name: Get server floating IP
+          set_fact:
+            osh_floating_ip: "{{ _osh_server_create['openstack']['accessIPv4'] }}"
+        # The run_ansible ensures the socok8s_workspace is not an empty var by
+        # exporting ANSIBLE_RUNNER_DIR
+        - name: Extend inventory in workspace with newly created SES node
+          config_template:
+            src: "{{ socok8s_workspace }}/inventory/hosts.yml"
+            dest: "{{ socok8s_workspace }}/inventory/hosts.yml"
+            config_type: yaml
+            config_overrides: "{{ osh_aio_overrides }}"
+          vars:
+            osh_aio_overrides:
+              osh-deployer:
+                hosts:
+                  osh-aio:
+                    ansible_host: "{{ osh_floating_ip }}"
+
+        - meta: refresh_inventory
+
+        # Waiting for port is not enough, as the port might be up (systemd)
+        # while ssh is not ready. Wait for connection is not good either
+        # because the key is unknown.
+        - name: Wait for a while for SSH to be properly booted up
+          wait_for:
+            timeout: 120
+
+        - name: Get pubkey, and add it to known hosts
+          shell: ssh-keyscan -H {{ osh_floating_ip | quote }} >> ${HOME}/.ssh/known_hosts
+          args:
+            executable: /bin/bash
+
+    - name: Handle deletion
+      when:
+        - (osh_node_delete | default('false')) | bool
+      block:
+        - name: Delete server
+          os_server:
+            cloud: "{{ deploy_on_openstack_cloudname }}"
+            delete_fip: yes
+            name: "{{ deploy_on_openstack_oshnode_server_name }}"
+            state: absent
+            terminate_volume: yes
```

Comment:
Doesn't `terminate_volume` delete a volume only if booted from volume?

Reply:
Yes indeed. I can probably remove it.

Reply:
FYI it doesn't matter to delete it, as the volume name would be always the same, which therefore is idempotent.

Reply:
But is it OK to use the volume with some old data on redeploy? I'd think it's better to delete it (correctly)

Reply:
I agree, will implement this. It's better to be reliable than earning a few seconds on delete.